daemon/daemon.go

```diff
@@ -250,15 +250,9 @@ func (d *Daemon) Dying() <-chan struct{} {
 	return d.tomb.Dying()
 }
 
-func (d *Daemon) auther(r *http.Request) (store.Authenticator, error) {
+func (d *Daemon) auther(user *auth.UserState) (store.Authenticator, error) {
```

Comment:
I agree that it's error-prone. The key requirement is that we need to be able to get an Authenticator with device credentials regardless of whether there is a user involved in the interaction, so keeping a DeviceState in the UserState doesn't currently work.

We could create a cred-less UserState for eg. local root, but maybe it's better to have the concept of a non-persistent anonymous UserState instead, with eg. ID=0. Only a handful of callsites that test anonymity would need to be updated. Does the anonymous UserState idea sound sensible?